lib/events.js

```diff
@@ -352,13 +352,30 @@ EventEmitter.prototype.emit = function emit(type, ...args) {
   return true;
 };
 
-function _addListener(target, type, listener, prepend) {
+function _addListener(target, type, listener, prepend, options) {
   let m;
   let events;
   let existing;
 
   checkListener(listener);
 
+  if (options !== undefined) {
+    const signal = options ? options.signal : undefined;
+    validateAbortSignal(signal, 'options.signal');
+    if (signal.aborted) {
```

Comment:
This would throw if `signal === undefined`, btw, and it doesn’t look like that would be intentional